src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1,4 +1,59 @@
 featurizers:
+  # ----------------------------------------------------------------------
+  - name: BackwardFillImputerFeaturizer
+    estimator_name: BackwardFillImputerEstimator
+    release_version: 0.3.6
+    num_output_columns: 1
+    creates_annotations: false
+    description: |-
+      Imputes (populates) previously encountered null values with the current non-null value encountered during
+      inference time.
+
+      C++-style pseudo signature:
+        float execute(float const &value);
+        std::string execute(std::optional<std::string> const &value);
+
+      Examples:
+        execute(1.0) -> 1.0
+
+        execute(NaN) -> 2.0  // Latest non-null value is 2.0
+        execute(2.0) -> 2.0
+
+        execute(3.0) -> 3.0
+
+        execute(NaN) -> 4.0  // Last non-null value was 4.0
+        execute(NaN) -> 4.0  // Last non-null value was 4.0
+        execute(4.0) -> 4.0
+
+        execute(NaN)
+        flush() // Exception, as pending values remain
+
+    has_dynamic_output: true
+    transformer_maintains_state: true
+
+    templates:
+      - name: T
+        types:
+          - int8

```

Comment:
I remember Dmitri suggested that it would be better to list these types sizewise, like int8, uint8, int16, uint16... Will there be another refactor for this?

Reply:
The kernels have been refactored already.

---
In reply to: [377926952](https://github.com/microsoft/FeaturizersLibrary/pull/139#discussion_r377926952) [](ancestors = 377926952)

Reply:
Gotcha!